packages/protocol/test-sol/common/GasPriceMinimum.t.sol

```diff
@@ -0,0 +1,272 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.7 <0.8.20;
+
+import "celo-foundry-8/Test.sol";
+
+import "../../contracts/common/FixidityLib.sol";
+
+import "../../contracts/common/interfaces/IRegistry.sol";
+
+// Contract to test
+import "../../contracts-0.8/common/GasPriceMinimum.sol";
+
+contract GasPriceMinimumTest is Test {
+  using FixidityLib for FixidityLib.Fraction;
+
+  IRegistry registry;
+  GasPriceMinimum public gasPriceMinimum;
+  address owner;
+  address nonOwner;
+
+  uint256 gasPriceMinimumFloor = 100;
+  uint256 initialGasPriceMinimum = gasPriceMinimumFloor;
+  uint256 targetDensity = FixidityLib.newFixedFraction(5, 10).unwrap();
+  FixidityLib.Fraction targetDensityFraction = FixidityLib.newFixedFraction(5, 10);
+  uint256 adjustmentSpeed = FixidityLib.newFixedFraction(5, 10).unwrap();
+  FixidityLib.Fraction adjustmentSpeedFraction = FixidityLib.newFixedFraction(5, 10);
+  address registryAddress = 0x000000000000000000000000000000000000ce10;
+
+  event TargetDensitySet(uint256 targetDensity);
+  event GasPriceMinimumFloorSet(uint256 gasPriceMinimumFloor);
+  event AdjustmentSpeedSet(uint256 adjustmentSpeed);
+  event GasPriceMinimumUpdated(uint256 gasPriceMinimum);
+  event BaseFeeOpCodeActivationBlockSet(uint256 baseFeeOpCodeActivationBlock);
+
+  function setUp() public virtual {
+    owner = address(this);
+    nonOwner = actor("nonOwner");
+
+    deployCodeTo("Registry.sol", abi.encode(false), registryAddress);
+    gasPriceMinimum = new GasPriceMinimum(true);
+
+    registry = IRegistry(registryAddress);
+
+    registry.setAddressFor("GasPriceMinimum", address(gasPriceMinimum));
+
+    gasPriceMinimum.initialize(
+      registryAddress,
+      gasPriceMinimumFloor,
+      targetDensity,
+      adjustmentSpeed,
+      0
+    );
+  }
+}
+
+contract GasPriceMinimumInitialize is GasPriceMinimumTest {
+  function test_shouldHaveSetOwner() public {
+    assertEq(gasPriceMinimum.owner(), owner);
+  }
+
+  function test_shouldSetTheGasPriceMinimum() public {
+    assertEq(gasPriceMinimum.getGasPriceMinimum(address(0)), initialGasPriceMinimum);
+  }
+
+  function test_shouldHaveTargetDensity() public {
+    assertEq(gasPriceMinimum.targetDensity(), targetDensity);
+  }
+
+  function test_shouldHaveAdjustmentSpeed() public {
+    assertEq(gasPriceMinimum.adjustmentSpeed(), adjustmentSpeed);
+  }
+
+  function test_shouldHaveGasPriceMinimumFloor() public {
+    assertEq(gasPriceMinimum.gasPriceMinimumFloor(), gasPriceMinimumFloor);
+  }
+
+  function test_shouldRevertWhenCalledAgain() public {
+    vm.expectRevert("contract already initialized");
+    gasPriceMinimum.initialize(
+      registryAddress,
+      gasPriceMinimumFloor,
+      targetDensity,
+      adjustmentSpeed,
+      0
+    );
+  }
+}
+
+contract GasPriceMinimumSetAdjustmentSpeed is GasPriceMinimumTest {
+  using FixidityLib for FixidityLib.Fraction;
```

Comment:
add this to parent?

Reply:
I believe that in `pragma ^0.8` it is required to specify `using A for B;` in each contract.

> The using A for B; directive is active only within the current scope (either the contract or the current module/source unit), including within all of its functions, and has no effect outside of the contract or module in which it is used.

https://docs.soliditylang.org/en/v0.8.15/contracts.html#using-for